h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -681,8 +685,9 @@ public final boolean hasNext() {
         @Override
         public final X next() {
             if(!hasNext()) {
-                return null;
+                throw new NoSuchElementException();
             }
+            assert current != null;
```

Comment:
this class should probably also init current in the constructor, then we can just do

    if (current == null) {
        throw new NoSuchElementException()
    }

here